### PR TITLE
Improvement/simulation

### PR DESCRIPTION
**Pull request by TimBuckers**
Features
- colors for the links in gazebo 
- collision in the feet
- default_planner_config warning fixed, None is not a tag should be ""
- make it easy to switch between fixed model in gazebo (airgaiting) and just falling down on the floor (standing)

closes https://github.com/project-march/march-iv/issues/141

### CODE REVIEW

File: march_control/launch/march_control.launch [hunk 1]
```diff
@@ -34,7 +35,8 @@
     <!-- Launch the gazebo package. Optionally with ui. -->
     <arg name="gazebo_ui" default="false" doc="Launch Gazebo with a ui"/>
     <include file="$(find march_gazebo)/launch/march_world.launch">
-        <arg name="gui" value="$(arg gazebo_ui)"/>
+        <arg name="gui" value="true"/>
```
Comment (Ishadijcks) [inline march_control/launch/march_control.launch]: Seems like a debug remainder

File: march_control/launch/march_control.launch [hunk 2]
```diff
@@ -12,9 +12,10 @@
     <arg name="debug" default="true"/>
 
     <!-- Load the joint controllers -->
-    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" ns="/march" output="screen" args="trajectory_controller joint_state_controller"/>
+    <node name="controller_spawner" pkg="controller_manager" type="spawner" ns="/march" output="screen"
+          args="trajectory_controller joint_state_controller"/>
 
-    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true"
+    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
```
Comment (Ishadijcks) [inline march_control/launch/march_control.launch]: Why was this needed?

File: march_description/urdf/march-iii.xacro [hunk 3]
```diff
@@ -1,6 +1,11 @@
 <?xml version="1.0"?>
 <!-- Revolute-Revolute Manipulator -->
 <robot name="march" xmlns:xacro="http://www.ros.org/wiki/xacro">
+    <!-- Switch between airgaiting or standing on the ground-->
+    <!--Value set to: 'world' fixes the robot to the world-->
+    <!--Value set to: 'base_link' makes the robot stand on the ground-->
+    <!--Important note: when changing value make sure rviz fixed frame is set to the same value (moveit.rviz)-->
+    <xacro:property name="parent_name" value="world"/>
```
Comment (Ishadijcks) [inline march_description/urdf/march-iii.xacro]: Can we change this with a URDF parameter instead of manually?

File: march_description/urdf/march-iii.xacro [hunk 4]
```diff
@@ -47,10 +52,10 @@
     </plugin>
 
     <!-- Used for fixing robot to Gazebo 'base_link' -->
-    <link name="world"/>
+    <link name="${parent_name}"/>
 
     <joint name="hip" type="fixed">
-        <parent link="world"/>
+        <parent link="${parent_name}"/>
```
Comment (Ishadijcks) [inline march_description/urdf/march-iii.xacro]: Clever fix :+1:

File: march_moveit/config/ompl_planning.yaml [hunk 5]
```diff
@@ -197,4 +201,6 @@ full_exo:
     - LazyPRM
     - LazyPRMstar
     - SPARS
-    - SPARStwo
\ No newline at end of file
+    - SPARStwo
+  projection_evaluator: joints(left_hip,left_knee)
```
Comment (Ishadijcks) [inline march_moveit/config/ompl_planning.yaml]: What is this exactly?

File: march_description/urdf/march-iii.xacro [hunk 6]
```diff
@@ -1,6 +1,11 @@
 <?xml version="1.0"?>
 <!-- Revolute-Revolute Manipulator -->
 <robot name="march" xmlns:xacro="http://www.ros.org/wiki/xacro">
+    <!-- Switch between airgaiting or standing on the ground-->
+    <!--Value set to: 'world' fixes the robot to the world-->
+    <!--Value set to: 'base_link' makes the robot stand on the ground-->
+    <!--Important note: when changing value make sure rviz fixed frame is set to the same value (moveit.rviz)-->
```
Comment (Ishadijcks) [inline march_description/urdf/march-iii.xacro]: Can you pass the same parameter to the rviz launchfile to make sure they are identical?

File: march_control/launch/march_control.launch (hunk 2 above)
Comment (TimBuckers) [reply march_control/launch/march_control.launch]: the default is respawn false, automatically restarting a node when crashing makes it harder to detect if the system launched correctly.

File: march_moveit/config/ompl_planning.yaml (hunk 5 above)
Comment (TimBuckers) [reply march_moveit/config/ompl_planning.yaml]: This was generated by moveit, when removing hip from the planning group.

File: march_control/launch/march_control.launch (hunk 2 above)
Comment (Ishadijcks) [reply march_control/launch/march_control.launch]: Sounds fair

File: march_description/urdf/march-iii.xacro (hunk 6 above)
Comment (JitskedeVries) [reply march_description/urdf/march-iii.xacro]: With the base_link setting, the controller is not able to follow the trajectory anymore. Thus gives an error. This is not a problem I think.

File: march_description/urdf/march-iii.xacro [hunk 7]
```diff
@@ -177,6 +193,11 @@
                     iyy="${mass / 12.0 * (lower_leg_height*lower_leg_height + width*width)}" iyz="0.0"
                     izz="${mass / 12.0 * (width*width + width*width)}"/>
         </inertial>
+        <collision>
+            <geometry>
+                <box size="${foot_height} ${width} ${width}"/>
+            </geometry>
+        </collision>
```
Comment (JitskedeVries) [inline march_description/urdf/march-iii.xacro]: Why do you give only the feet the collision property? In reality every part has collision, right?

File: march_moveit/config/march.srdf [hunk 8]
```diff
@@ -13,24 +13,17 @@
...
         <joint name="right_knee" />
         <joint name="right_ankle" />
-        <joint name="hip" />
     </group>
     <group name="full_exo">
         <group name="left_leg" />
         <group name="right_leg" />
     </group>
     <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
-    <group_state name="home_sit" group="left_leg">
```
Comment (JitskedeVries) [inline march_moveit/config/march.srdf]: Why is this state being deleted?

File: march_description/urdf/march-iii.xacro (hunk 7 above)
Comment (Ishadijcks) [reply march_description/urdf/march-iii.xacro]: Collision is only activated in the feet as those are the only parts of the exoskeleton interacting with the world. It's currently disabled as it is a slight performance boost.

However it can look weird when the exoskeleton falls through the ground except for the feet. As self-collision is disabled anyway I don't think the performance would matter much @TimBuckers ?

File: march_moveit/config/march.srdf (hunk 8 above)
Comment (TimBuckers) [reply march_moveit/config/march.srdf]: This was for the planning group 'left_leg' only, all the poses at this point are for the group 'full_exo'

File: march_description/urdf/march-iii.xacro (hunk 7 above)
Comment (TimBuckers) [reply march_description/urdf/march-iii.xacro]: The visual size of most links is guessed, because the link shapes are only to make the movement visible. At some point I think is nice to improve the shape of the collision of the feet, if we want to create a more realistic simulation.

File: march_description/urdf/march-iii.xacro (hunk 3 above)
Comment (TimBuckers) [reply march_description/urdf/march-iii.xacro]: Perhaps, but we are not going to switch this often. First we have to determine what we want to do with a simulation in general.
